sycl/source/detail/device_image_impl.hpp

```diff
@@ -167,7 +217,9 @@ class device_image_impl {
   // image
   std::vector<unsigned char> MSpecConstsBlob;
   // Contains list of spec ID + their offsets in the MSpecConstsBlob
-  std::vector<SpecConstDescT> MSpecConstDescs;
+  // std::vector<SpecConstDescT> MSpecConstDescs;
+
+  std::map<const char *, std::vector<SpecConstDescT>> MSpecConstSymMap;
```

Comment:
```suggestion
  std::map<std::string, std::vector<SpecConstDescT>> MSpecConstSymMap;
```
For checking string by value rather than by its pointer.